README
NAME
    YamlTime - YAML-based Personal Time Tracking App

SYNOPSIS
        > yt help

DESCRIPTION
    YamlTime is an application that allows you do your personal project time
    tracking from the command line. It saves your data in plain text YAML
    files. You can use a version control system (like git) to back up the
    data.

    YamlTime can supports multiple customers, multiple projects and multiple
    rates. It has reporting, spreadsheets, tagging, time/tag selection and
    3rd party plugin modules.

    YamlTime comes with a command line app called "yt" that does everything.

COMMAND LINE USAGE
    The following commands are supported.

        yt                  - Show current yt status of today's tasks
        yt help             - Get help
        yt help <command>   - Get help for a specific command
        yt init             - Create a new YamlTime store
        yt new              - Start a new task
        yt stop             - Stop the current task
        yt go               - Restart the current task
        yt create <task>    - Create a new task for a specific date/time
        yt edit <task>      - Edit a task's yaml file in $EDITOR
        yt dump <task>      - Read a task file and print to STDOUT
        yt delete <task>    - Delete a task file
        yt check <tasks>    - Check the data in the range for errors
        yt status <tasks>   - Show the current yt status
        yt report <tasks>   - Create a report for a time period
                                using a certain reporting style

    The <yt new> command will prompt you for some information. You can use
    tab completion for many of the fields. The values that you put in your
    config files are the values that are offered (and the only ones you can
    use).

    Some of the fields may be left blank. You can use "yt edit" to fix up
    any task, later on.

  Options
    yt commands have the following options:

    "--from=<date_string>"
        Commands that need a time range, use this to set the start time. The
        default is the previous midnight. A human friendly string can be
        used, like: '3 days ago'.

    "--to=<date_string>"
        Commands that need a time range, use this to set the end time. The
        default is now.

    "--tag=<tag_list>"
        A comma separated list of tags. Matches tasks the match all the
        tags. You can specify more than once to combine ('or' logig) groups.

    "--style=<report-style>"
        This names a YamlTime reporting style. The default is CSV, which can
        be used as a spreadsheet.

KUDOS
    Many thanks to the good people of Strategic Data in Melbourne Victoria
    Australia, for supporting me and this project. o/

AUTHOR
    Ingy döt Net <ingy@cpan.org>

COPYRIGHT
    Copyright 2011-2014. Ingy döt Net.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.

    See <http://www.perl.com/perl/misc/Artistic.html>



Changes
0.18 Sat Aug 16 13:07:19 PDT 2014
 - Meta 0.0.2

0.17 Sat Aug 16 04:30:49 PDT 2014
 - Eliminate spurious trailing whitespace

0.16 Fri Aug 15 21:34:45 PDT 2014
 - Remove t/use_ok.t

0.15 Fri Aug 15 21:14:24 PDT 2014
 - Add t/000-compile-modules.t

0.14 Sat Aug  9 00:34:08 PDT 2014
 - Fix swim errors

0.13 Fri Aug  8 13:01:04 PDT 2014
 - Switch hashbang to '#!/usr/bin/env perl'

0.12 Tue Aug  5 11:40:36 PDT 2014
 - Switch to Zilla-Dist

0.11 Tue Oct 18 07:44:17 CDT 2011
 - Make YAMLTIME_DEFAULT_COMMAND work.

0.10 Thu Jul 28 14:22:37 CDT 2011
 - Load plugins from 'YamlTime::' namespace instead of YamlTime::Extension
 - Support $YAMLTIME_DEFAULT_COMMAND
 - Support ~/.yamltime/

0.09 Fri Jul  8 00:18:25 EDT 2011
 - Add base command
 - Fix 'go' math for adding hours/mins
 - Fix 'delete' command

0.08 Mon Jun 27 21:41:45 EDT 2011
 - 0.07 was broken. add test, and make it work
 - Now support YamlTime::Extension::Foo

0.07 Mon Jun 27 08:59:13 EDT 2011
 - rjbs++' changes

0.06 Tue Jun 14 22:35:45 EST 2011
 - Add better readline support and tab completion

0.05 Sat Jun 11 16:22:00 EST 2011
 - Add tag support to ranges
 - Refactor error msgs
 - Remove $HOME/.yt as a default. You can set YAMLTIME_BASE to it.

0.04 Sat Jun 11 03:16:02 EST 2011
 - Polish the help msgs
 - Refactoring

0.03 Fri Jun 10 01:42:52 EST 2011
 - Finish the rest of the commands
 - Fix extra blank line in yaml files
 - Add dump action
 - Add remove action
 - Refactor default_command
 - Better error msgs
 - Check 'project' better.

0.02 Wed Jun  8 18:47:05 EST 2011
 - Fixed a newline down under.

0.01 Wed Jun  8 18:43:36 EST 2011
 - First Release



Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.019.
use strict;
use warnings;

use 5.008001;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "YAML-based Personal Time Tracking App",
  "AUTHOR" => "Ingy d\x{f6}t Net <ingy\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "YamlTime",
  "EXE_FILES" => [
    "bin/yt"
  ],
  "LICENSE" => "perl",
  "NAME" => "YamlTime",
  "PREREQ_PM" => {
    "App::Cmd::Setup" => 0,
    "Capture::Tiny" => "0.11",
    "DateTime" => "0.70",
    "DateTime::Format::Natural" => "0.96",
    "File::ShareDir" => "1.03",
    "IO::All" => "0.43",
    "Module::Pluggable" => 0,
    "Mouse" => "0.93",
    "MouseX::App::Cmd" => "0.08",
    "Template::Plugin::YAMLVal" => "0.10",
    "Template::Toolkit::Simple" => "0.13",
    "Term::ReadLine" => "1.07",
    "Term::ReadLine::Gnu" => "1.20",
    "Text::CSV_XS" => "0.82",
    "Text::ParseWords" => "3.27",
    "YAML::XS" => "0.35"
  },
  "VERSION" => "0.18",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "App::Cmd::Setup" => 0,
  "Capture::Tiny" => "0.11",
  "DateTime" => "0.70",
  "DateTime::Format::Natural" => "0.96",
  "File::ShareDir" => "1.03",
  "IO::All" => "0.43",
  "Module::Pluggable" => 0,
  "Mouse" => "0.93",
  "MouseX::App::Cmd" => "0.08",
  "Template::Plugin::YAMLVal" => "0.10",
  "Template::Toolkit::Simple" => "0.13",
  "Term::ReadLine" => "1.07",
  "Term::ReadLine::Gnu" => "1.20",
  "Text::CSV_XS" => "0.82",
  "Text::ParseWords" => "3.27",
  "YAML::XS" => "0.35"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


